frame/executive/src/lib.rs

```diff
@@ -355,6 +357,13 @@ where
 
 		// post-extrinsics book-keeping
 		<frame_system::Module<System>>::note_finished_extrinsics();
+		//TODO: does this need to be implemented here too?
+		// let weight =  frame_system::BlockWeight;
+		// debug::info!("book keeping {}", weight);
+
+		// <frame_system::Module<System> as OnIdle<System::BlockNumber>>::on_idle(block_number);
+		// <AllModules as OnIdle<System::BlockNumber>>::on_idle(block_number);
+
```

Comment:
Maybe @bkchr can answer that. Not sure why the `on_finalize` code is duplicated.

Reply:
`finalize_block` is called by block producer after they apply all extrinsic they wanted.
`execute_extrinsic_with_book_keeping` is executed inside `execute_block`.
Note that `finalize_block` do call `system::finalize` while `execute_extrinsics_with_book_keeping` doesn't (instead it is execute in final_checks)

But it seems it can be factorize.